src/jansson.h

```diff
@@ -50,6 +52,7 @@ typedef enum {
 typedef struct json_t {
     json_type type;
     size_t refcount;
+    pthread_mutex_t refmutex;
```

Comment:
We could have a single mutex for all `json_incref`/`json_decref` calls. I don't think it would be absolutely terrible, but it would definitely be a bottleneck.

This absolutely calls for atomics; I just wasn't up for figuring out a safe atomic algorithm last night. Or today. Or perhaps most days.